book/src/getting-started/2-our-first-cxx-qt-module.md

```diff
@@ -31,49 +32,57 @@ This is a lot to take in, so let's go one step at a time.
 In our case that means:
 - `number: i32` -> `int number`
-- `string: QString` -> `QString string`\
+- `string: QString` -> `QString string`
+
+For `i32`, CXX already knows how to translate it.
+A `QString` however is unknown to CXX.
+Luckily, the `cxx_qt_lib` crate already wraps many Qt types for us.
+We can just import them like any other CXX type:
+``` rust, ignore
+{{#include ../../../examples/qml_minimal/rust/src/cxxqt_object.rs:book_qstring_import}}
+```
 For more details on the available types, see the [Qt types page](../concepts/types.md).
 
-You might have also noticed the `impl Default for MyObject` here.
-The struct needs to always be default-constructable and allows you to provide default values for your `Q_PROPERTY`s.
+CXX-Qt will then automatically generate a new QObject subclass for our `MyObject` struct and expose it as an [`extern "C++"` opaque type](https://cxx.rs/extern-c++.html#opaque-c-types) to Rust.
+For any Rust struct `T` that is marked with `#[cxx_qt::qobject]`, CXX-Qt will expose its QObject wrapper under `qobject::T`.
+In our case, this means we can refer to the QObject wrapper for our `MyObject` struct, as `qobject::MyObject`.
 
-The name of this struct is used as the name of the C++ QObject subclass, in our case this is struct only contains properties.
-However, the `#[cxx_qt::qobject]` marked struct could contain any fields we want which are Rust only.
-
-Just because the `#[cxx_qt::qobject]` marked struct struct doesn't contain any data, that still doesn't mean its not an important part of our `MyObject` class.
-That is because it actually defines the behavior of our class through its `impl`:
+This type can be used like any other CXX opaque type.
+Additionally, CXX-Qt allows us to add functionality to this QObject by using `impl qobject::MyObject` together with `#[qinvokable]`.
 ```rust,ignore
 {{#include ../../../examples/qml_minimal/rust/src/cxxqt_object.rs:book_rustobj_impl}}
 ```
 
 In our case, we define two new functions:
 - `increment_number`
     - Increments the number of the `MyObject`.
-    - As the number lives on the C++ side, it uses a `CppObj` wrapper which is generated by CXX-Qt and has the appropriate setters and getters for each property.
     - The name will be converted to `incrementNumber` in C++.
 - `say_hello`
     - Prints a provided number and string.
     - The name will be converted to `sayHello` in C++.
 
-Both functions are marked with the `#[qinvokable]` macro, which means the functions will be added to the C++ code of `MyObject` and will be callable from QML as well.
-
-Apart from functions marked with the `#[qinvokable]` macro, the `#[cxx_qt::qobject]` marked struct impl is just a normal Rust struct impl and can contain normal Rust functions, which the invokable functions can call as usual.
+Apart from functions marked with the `#[qinvokable]` macro, you can also define normal helper functions on this struct that won't be exposed to QML.
```

Comment:
We also have a third one, where you don't use `#[qinvokable]` and imp on the qobject::T. These are C++ context but not added in the C++ definition, don't know if we want to introduce these here as they are kinda advanced.

But I think somewhere here it should be mentioned that when you put methods in `impl qobject::T` that the `self` is in the context of the wrapper / C++ class and not in the context of the Rust struct etc.

Reply:
Ah, hm, that was actually what I wanted to describe with this sentence.
But obviously I've failed xD.
So let me try to rephrase it.

Reply:
Also, I wanted to skip `impl MyObject` for now.
That part could be pretty confusing for newcomers.
They should stick to `impl qobject::MyObject` at first.

Reply:
Maybe just keep the bit how you mention you can impl Rust context methods as normal on the Rust struct as a side comment.

I think the main thing to focus on here though is how `impl qobject::T` the `self` is actually the C++ context not the Rust context, so to access Rust things you need to go through our getter/setter or the rust/rust_mut stuff